### PR TITLE
Use Path instead of URI for artifact, datasets and log meta

### PR DESCRIPTION
**Pull request by deepakwadhwani**

### CODE REVIEW

**Comment by deepakwadhwani** [comment]
Besides one test failure [DatasetInstanceHandlerTest.testBasics] that I removed from the brach, build is clean: http://builds.cask.co/browse/CDAP-DUT5328-16

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetTypeHandlerTest.java [hunk 1]
```diff
@@ -281,8 +282,9 @@ static void verify(DatasetModuleMeta moduleMeta, String moduleName, Class module
     Assert.assertArrayEquals(Sets.newTreeSet(usedByModules).toArray(),
                              Sets.newTreeSet(moduleMeta.getUsedByModules()).toArray());
     // note: we know it is local
-    Assert.assertNotNull(moduleMeta.getJarLocation());
-    Assert.assertTrue(new File(moduleMeta.getJarLocation()).exists());
```
Comment (albertshau) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetTypeHandlerTest.java:285]: why doesn't this work, isn't it the jar path an absolute path?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java [hunk 2]
```diff
@@ -1100,47 +1096,66 @@ private ArtifactCell(Id.Artifact artifactId) {
...
 
     ArtifactData(Location location, ArtifactMeta meta) {
       this.locationURI = null;
-      this.locationPath = Locations.getRelativePath(location.getLocationFactory(), location.toURI());
+      this.locationPath = location.toURI().getPath();
       this.meta = meta;
     }
+
+    public String getLocationPath() {
+      if (locationPath == null) {
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java]: return locationPath == null ? locationURI.getPath() : locationPath

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java [hunk 3]
```diff
@@ -372,7 +373,7 @@ public Boolean call() throws DatasetModuleConflictException, IOException {
               impersonator.doAs(datasetModuleId.getParent(), new Callable<Location>() {
                 @Override
                 public Location call() throws Exception {
-                  return locationFactory.create(module.getJarLocation());
+                  return Locations.getLocationFromAbsolutePath(locationFactory, module.getJarLocationPath());
```
Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java:376]: users of the DatasetModuleMeta class should not have to know they have to use this utility method. You should put a `getLocation(LocationFactory)` method in DatasetModuleMeta that handles the logic, then these classes call that method.

File: cdap-proto/src/main/java/co/cask/cdap/proto/DatasetModuleMeta.java [hunk 4]
```diff
@@ -32,6 +32,7 @@
   private final String className;
   // TODO: change type to Location
```
Comment (albertshau) [inline cdap-proto/src/main/java/co/cask/cdap/proto/DatasetModuleMeta.java:33]: comment that the URI is here for backwards compatibility

File: cdap-proto/src/main/java/co/cask/cdap/proto/DatasetModuleMeta.java [hunk 5]
```diff
@@ -91,10 +93,15 @@ public String getClassName() {
...
    * @return location of the dataset module jar, {@code null} means this is "system module" which classes always present
    *         in classpath. This helps to minimize redundant copying of jars
+   *         For Backward Compatibility, if DatasetModuleMeta was read from the table and only has jarLocation URI,
+   *         get the Path from URI and return.
    */
   @Nullable
-  public URI getJarLocation() {
-    return jarLocation;
+  public String getJarLocationPath() {
+    if (jarLocationPath == null && jarLocation != null) {
```
Comment (albertshau) [inline cdap-proto/src/main/java/co/cask/cdap/proto/DatasetModuleMeta.java]: add a comment that this check is for backwards compatibility

**Review by albertshau — COMMENTED** [review]
minor comments and one question

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java (hunk 3 above)
Comment (deepakwadhwani) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java:376]: Adding dependency on module 'cdap-common' in cdap-proto introduces circular dependency between modules cdap-security-spi and cdap-proto.

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**